src/ray/gcs/gcs_client/global_state_accessor.cc

```diff
@@ -68,6 +68,8 @@ std::vector<std::string> GlobalStateAccessor::GetAllJobInfo() {
     absl::ReaderMutexLock lock(&mutex_);
     RAY_CHECK_OK(gcs_client_->Jobs().AsyncGetAll(
         TransformForMultiItemCallback<rpc::JobTableData>(job_table_data, promise),
+        /*skip_job_info_field=*/false,
+        /*skip_is_running_tasks_field=*/false,
```

Comment:
Can those be True? Are those two fields used?

Reply:
`GlobalStateAccessor::GetAllJobInfo()` is invoked by multiple functions, including

https://github.com/ray-project/ray/blob/f40313b0acc50a60957d1ecd2b75946ca6079ce4/python/ray/_private/state.py#L170

which explicitly sets two params to True as it doesn't utilize those fields.

And

https://github.com/ray-project/ray/blob/f40313b0acc50a60957d1ecd2b75946ca6079ce4/java/runtime/src/main/java/io/ray/runtime/gcs/GlobalStateAccessor.java#L50

may be called by external methods, so we're leaving it unchanged and defaulting to False.

To accommodate these varying callers, have added explicit `skip` params to `GetAllJobInfo()`.